decidim-admin/app/forms/decidim/admin/component_form.rb

```diff
@@ -90,6 +90,8 @@ def budget_voting_rule_enabled_setting
 
       # - the value must be a valid number
       def budget_voting_rule_value_setting
+        return unless manifest&.name == :budgets
```

Comment:
I think we should try to separate this logic and extract it from here, so that we don't link the admin component with the budgets one. 

It would require a little refactor, but what about making the component manifest define an optional class, something like `MyComponent::ComponentSettingsForm` that receives the settings and validates them? The `Admin::ComponentForm` would find this class via the component manifest, call it if present and check if it's valid. If it's not present, then we go ahead. If it's present and valid, then go ahead. If it's present and not valid, we return the errors. This would open the possibility for other components to implement extra logic on settings.

What do you think? Would that work for you?

Reply:
Yes, I like the idea, I'll refactor it. Thanks @mrcasals 😃